### ISSUE TITLE
tf.gradients() in phi_symbolic()of TFDeepExplainer returns None

### ISSUE THREAD

**Issue by changyc14** [shap/shap#653]
I use tensorflow.python.estimator to train a model. Then I pass the session and graph to DeepExplainer, and inside phi_symbolic() where `tf.gradients(out, self.model_inputs)` returns [None]. I am not sure where goes wrong. Can you please provide some guidance? Thanks!

Here's a snippet of code I modify in phi_symbolic():
```
try:
    with self.graph.as_default():
        out = self.model_output[:, i] if self.multi_output else self.model_output
        self.phi_symbolics[i] = tf.gradients(out, self.model_inputs)
```

I try to reproduce the error by not using shap and print critical elements to get tf.gradients, and I get valid gradient tensors as shown below.
graph: `<tensorflow.python.framework.ops.Graph object at 0x10615bc50>`
model_inputs: `[<tf.Tensor 'Placeholder:0' shape=(?, 12) dtype=float32>, <tf.Tensor 'Placeholder_1:0' shape=(?, 12) dtype=float32>, <tf.Tensor 'Placeholder_2:0' shape=(?, 12) dtype=float32>]`
out: `Tensor("strided_slice:0", shape=(?,), dtype=float32)`
phi_symbolics[i]: `[<tf.Tensor 'gradients/lstm/input_from_feature_columns/input_layer/var_0/Reshape_grad/Reshape:0' shape=(?, 12) dtype=float32>, <tf.Tensor 'gradients/lstm/input_from_feature_columns/input_layer_1/var_1/Reshape_grad/Reshape:0' shape=(?, 12) dtype=float32>, <tf.Tensor 'gradients/lstm/input_from_feature_columns/input_layer_2/var_2/Reshape_grad/Reshape:0' shape=(?, 12) dtype=float32>]`

Here is the result I get by using shap:
graph: `<tensorflow.python.framework.ops.Graph object at 0x10615bc50>`
model_inputs: `[<tf.Tensor 'Placeholder:0' shape=(?, 12) dtype=float32>, <tf.Tensor 'Placeholder_1:0' shape=(?, 12) dtype=float32>, <tf.Tensor 'Placeholder_2:0' shape=(?, 12) dtype=float32>]`
out: `Tensor("strided_slice_34:0", shape=(?,), dtype=float32)`
phi_symbolics[i]: `[None, None, None]`

If the graphs and the parameters of `tf.gradients()` are the same, why do I get different results? Is there anything in shap I need to take care of in order to make it right?

(tensorflow version: 1.8.0)

**Comment by slundberg** [comment]
I just pushed a fix that might have impacted this. Could you verify this problem still happens for you with SHAP 0.29.3? Thanks!

**Comment by changyc14** [comment]
I try SHAP 0.29.3, and the problem is still there.
